File: src/main/java/no/fint/ResourceConverter.java
package no.fint;

import com.fasterxml.jackson.databind.DeserializationFeature;
import com.fasterxml.jackson.databind.ObjectMapper;
import no.fint.model.administrasjon.organisasjon.Organisasjonselement;
import no.fint.model.resource.administrasjon.organisasjon.OrganisasjonselementResource;

import java.io.IOException;

public class ResourceConverter {
    public static Organisasjonselement toOrganisasjonselement(OrganisasjonselementResource resource) throws IOException {
        ObjectMapper objectMapper = new ObjectMapper()
                .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
        String asString = objectMapper.writeValueAsString(resource);
        return objectMapper.readValue(asString, Organisasjonselement.class);
    }
}

File: src/main/kotlin/no/fint/utils/ResourceConverter.kt
package no.fint.utils

import com.fasterxml.jackson.databind.DeserializationFeature
import com.fasterxml.jackson.databind.ObjectMapper
import no.novari.fint.model.administrasjon.organisasjon.Organisasjonselement
import no.novari.fint.model.resource.administrasjon.organisasjon.OrganisasjonselementResource

object ResourceConverter {
    private val objectMapper =
        ObjectMapper().apply {
            configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
        }

    fun toOrganisasjonselement(resource: OrganisasjonselementResource): Organisasjonselement {
        val asString = objectMapper.writeValueAsString(resource)
        return objectMapper.readValue(asString, Organisasjonselement::class.java)
    }
}
